web/assets/js/main_out.js

```diff
@@ -1181,63 +1214,72 @@
         return (a.x - b.x) * (a.x - b.x) + (a.y - b.y) * (a.y - b.y);
     }
     function updateQuadtree() {
-        var w = 1920 / camera.sizeScale;
-        var h = 1080 / camera.sizeScale;
-        var x = (camera.x - w / 2);
-        var y = (camera.y - h / 2);
-        quadtree = new PointQuadTree(x, y, w, h, QUADTREE_MAX_POINTS);
-        for (var i = 0; i < cells.list.length; ++i) {
-            var cell = cells.list[i];
-            for (var n = 0; n < cell.points.length; ++n) {
-                quadtree.insert(cell.points[n]);
+        const w = 1920 / camera.sizeScale;
+        const h = 1080 / camera.sizeScale;
+        const x = (camera.x - w / 2);
+        const y = (camera.y - h / 2);
+        quadtree = new window.PointQuadTree(x, y, w, h, QUADTREE_MAX_POINTS);
+        for (let i = 0; i < cells.list.length; ++i) {
+            const cell = cells.list[i];
+            for (let j = 0; j < cell.points.length; ++j) {
+                quadtree.insert(cell.points[j]);
             }
         }
     }
 
-    function Cell(id, x, y, s, name, color, skin, flags) {
-        this.id = id;
-        this.x = this.nx = this.ox = x;
-        this.y = this.ny = this.oy = y;
-        this.s = this.ns = this.os = s;
-        this.setColor(color);
-        this.setName(name);
-        this.setSkin(skin);
-        this.jagged = flags & 0x01 || flags & 0x10;
-        this.ejected = !!(flags & 0x20);
-        this.born = syncUpdStamp;
-        this.points = [];
-        this.pointsVel = [];
-    }
-    Cell.prototype = {
-        destroyed: false,
-        id: 0, diedBy: 0,
-        ox: 0, x: 0, nx: 0,
-        oy: 0, y: 0, ny: 0,
-        os: 0, s: 0, ns: 0,
-        nameSize: 0, drawNameSize: 0,
-        color: "#FFF", sColor: "#E5E5E5",
-        skin: null, jagged: false,
-        born: null, updated: null, dead: null, // timestamps
-        destroy: function(killerId) {
-            delete cells.byId[this.id];
-            if (cells.mine.remove(this.id) && cells.mine.length === 0)
-                showESCOverlay();
+    class Cell {
+        static parseName(value = '') { // static method
```

Comment:
Bug here, default arguments only work if the argument is `undefined`, but for ejected mass the name is `null`.

Reply:
You could probably change the `null` to `undefined` in the `wsMessage` function.